packages/tx/src/eip1559Transaction.ts

```diff
@@ -206,6 +206,21 @@ export class FeeMarketEIP1559Transaction extends BaseTransaction<TransactionType
     return EIP2930.getDataFee(this)
   }
 
+  /**
+   * Returns the minimum of calculated priority fee (from maxFeePerGas and baseFee) and maxPriorityFeePerGas
+   *
+   * * @param baseFee Base fee retrieved from block
+   */
+  getEffectivePriorityFee(baseFee: bigint | undefined): bigint {
```

Comment:
Feels like this could be a generic utility function rather than having to expand the class API or else a capability in eip1559.  I dunno, just feels like we have a lot of duplicate code across the txn classes.  I understand that legacy txns have a slightly different process to identify the effective tip so happy to let this sit as is if you feel its a cleaner approach.

Reply:
I think I can clean this up a bit :) I agree that this bloats the code.

Reply:
Have addressed